Source/Core/InputCommon/ControllerInterface/evdev/evdev.cpp

```diff
@@ -25,6 +25,56 @@
 
 namespace ciface::evdev
 {
+// Unfortunately there doesn't seem to be a "standard" orientation
+// so we can't use "Accel Up"-like names.
+static constexpr std::array<const char*, 6> s_motion_data_names = {{
+    "Accel X",
+    "Accel Y",
+    "Accel Z",
+    "Gyro X",
+    "Gyro Y",
+    "Gyro Z",
+}};
+
+class MotionDataInput : public Core::Device::Input
```

Comment:
This class looks very similar to Axis (in particular half of the ctor and GetState). Is it possible to reuse some of the functions from there?

Reply:
I've now eliminated the redundant `GetState` logic and data members with some base classes.
The similar calculation of `m_base` remains but I don't think it's worth worrying about.